### PR TITLE
Better jsx support

### PR DESCRIPTION
**Pull request by deanius**
Utilizing the `detective` npm module.
Issue #61 

Incidentally - why do npm modules such as `react` come out of the resolve function as dotted filesystem paths - is that correct behavior?

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,5 @@ node_modules
 !test/files/cjs/node_modules
 /.project
 /.settings
+
+.vscode/launch.json
```
Comment (mrjoelkemp) [inline .gitignore]: This should be in your global gitignore since it's not a file that's part of the project.

File: package.json [hunk 2]
```diff
@@ -32,6 +32,7 @@
     "detective": "^4.2.0",
     "detective-es6": "1.1.2",
     "graphviz": "0.0.8",
+    "precinct": "^2.5.1",
```
Comment (mrjoelkemp) [inline package.json]: Prefer upgrading detective-es6@1.1.3. That's all we really want from precinct anyway (we don't need the css or amd/cjs detectives to be included).

File: lib/parse/base.js [hunk 3]
```diff
@@ -159,8 +159,6 @@ Base.prototype.getFileSource = function (filename) {
 			header: false,
 			bare: true
 		});
-	} else if (filename.match(this.jsxExtRegEx)) {
```
Comment (mrjoelkemp) [inline lib/parse/base.js]: Does this need to be removed?

File: .jshintrc [hunk 4]
```diff
@@ -5,6 +5,7 @@
 	"jquery" : false,
 	"couch" : false,
 	"es5" : false,
+	"esnext": true,
```
Comment (mrjoelkemp) [inline .jshintrc]: Why is this necessary?

File: package.json (hunk 2 above)
Comment (deanius) [inline package.json]: That didn't do the trick when I tried it

File: lib/parse/base.js (hunk 3 above)
Comment (deanius) [inline lib/parse/base.js]: It caused the file to be transpiled, and that's where exceptions about `Illegal import declaration` were raised. Using detective/precinct theres no need for that transpiling

File: .jshintrc (hunk 4 above)
Comment (deanius) [inline .jshintrc]: Because I wrote tests in ES6, because— the future :)

Comment (mrjoelkemp) [inline .jshintrc]: Since the rest of the test suite is in ES5, it might make sense to retain uniformity.

File: lib/parse/base.js (hunk 3 above)
Comment (mrjoelkemp) [inline lib/parse/base.js]: To clarify, you're saying that the call to `jsx.transform` throws errors? Is that an issue with the `react-tools` library?

File: package.json (hunk 2 above)
Comment (mrjoelkemp) [inline package.json]: Precinct only [uses a caret for detective-es6](https://github.com/mrjoelkemp/node-precinct/blob/master/package.json#L31) (effectively upgrading 1.1.3), [sniffs the module type](https://github.com/mrjoelkemp/node-precinct/blob/master/index.js#L45), and then [delegates to detective-es6](https://github.com/mrjoelkemp/node-precinct/blob/master/index.js#L57). 

Try running `npm install --save detective-es6@1.1.3` in your madge fork and seeing if that works.

Comment (deanius) [inline package.json]: Interesting. I'll try that.

File: lib/parse/base.js (hunk 3 above)
Comment (deanius) [inline lib/parse/base.js]: Yes, should probably be removed. And `react-tools` is deprecated per: https://facebook.github.io/react/blog/2015/06/12/deprecating-jstransform-and-react-tools.html.

Incidentally, are you a maintainer on `madge`, or - what's your interest in it?

Comment (deanius) [inline lib/parse/base.js]: react-tools removed, and the detective 1.1.3 did what it should :)

Comment (mrjoelkemp) [inline lib/parse/base.js]: > Incidentally, are you a maintainer on madge, or - what's your interest in it?

Just a fan. I also like seeing if there are any problems with the detective usage.

> react-tools removed

I don't have enough context to know whether or not it's safe to remove this. cc @pahen

**Comment by pahen** [comment]
I suppose this can be closed now since https://github.com/pahen/madge/pull/83 is merged. Can you try the new release 0.5.4 @deanius and see if things are working as expected now?

**Comment by pahen** [comment]
This is fixed in the master branch now.
